charms/worker/k8s/lib/charms/k8s/v0/k8sd_api_manager.py

```diff
@@ -372,20 +372,38 @@ def _send_request(
         except (socket.error, HTTPException) as e:
             raise K8sdConnectionError(f"HTTP or Socket error: {e}") from e
 
-    def create_join_token(self, name: str):
+    def create_join_token(self, name: str, worker: bool = False):
         """Create a join token.
 
         Args:
             name (str): Name of the node.
+            worker (bool): Whether the node should join as control-plane or worker.
 
         Returns:
             str: The generated join token if successful.
         """
-        endpoint = "/cluster/1.0/tokens"
-        body = {"name": name}
+        endpoint = "/1.0/k8sd/cluster/tokens"
+        body = {
+            "name": name,
+            "worker": worker,
+        }
+
         join_response = self._send_request(endpoint, "POST", CreateJoinTokenResponse, body)
         return join_response.token
 
+    def join_cluster(self, name: str, address: str, token: str):
+        """Join a node to the k8s cluster.
+
+        Args:
+            name (str): Name of the node.
+            address (str): Address of the node.
```

Comment:
is this similar to `bootstrap_k8s_snap` in the same library?

```suggestion
            address: address to which k8sd should be bound
```